[Linus] iP

linuschui opened the pull request:
# DukePro
> "if your back hurts when you wake up in the morning, then wake up in the afternoon" - Sun Tzu, Art of War 2

DukePro frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://github.com/linuschui/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

FEATURES
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

linuschui committed:
Level-1

linuschui committed:
Level-1

linuschui committed:
Level-2

linuschui committed:
Level-2

linuschui committed:
Level-3

linuschui committed:
Merge remote-tracking branch 'origin/master'

linuschui committed:
Level-4

linuschui committed:
A-TextUiTesting

linuschui committed:
Level-5

linuschui committed:
Level-5

linuschui committed:
Level-5

linuschui committed:
Level-5

linuschui committed:
Level-6

linuschui committed:
Level-6

linuschui committed:
Merge remote-tracking branch 'origin/master'

linuschui committed:
Level-7

linuschui committed:
branch-Level-8

linuschui committed:
added tasks.txt

linuschui committed:
Fixed bugs from Level-7 when reading dates

linuschui committed:
A-MoreOOP

linuschui committed:
A-Packages

linuschui committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Add Gradle Support

linuschui committed:
A-JUnit

linuschui committed:
branch-A-JavaDoc

linuschui committed:
branch-Level-9

linuschui committed:
Level-9

linuschui committed:
Level-9

linuschui committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Duke/TaskList.java

linuschui committed:
Merged branch-Level-9

linuschui committed:
Fixed merge conflict and added JUnit tests for Level-9

linuschui committed:
Fixed issues with DukeException

ee-suan commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    /**
+     * builds a string based on the list of tasks contained in the ArrayList of Task objects
```
Maybe we should capitalise the first word (Builds)

ee-suan commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,59 @@
+    protected String description;
+
+    /**
+     * The done boolean decides if a task has been completed.
```
Maybe you mean isDone boolean?

ee-suan commented on src/test/java/Duke/FindTest.java:
```diff
@@ -0,0 +1,2 @@
+package Duke;public class FindTest {
```
Maybe a line break after the package statement? (Same for the other tests too)

ee-suan reviewed (commented):
Great work! I like the naming of methods and fields and the Javadocs

linuschui committed:
A-Checkstyle

linuschui committed:
Delete tasks

LolfoollorS commented on src/main/java/Command/Command.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * The exit boolean decides if the command exits the program.
+     */
+    private boolean exit;
```
Perhaps you could rename the variable to make it sound like a boolean.

LolfoollorS commented on src/main/java/Command/Command.java:
```diff
@@ -0,0 +1,41 @@
+*/
+public abstract class Command {
+
+    /**
+     * The exit boolean decides if the command exits the program.
+     */
```
May I ask if you intend to write it as comment/comment block or as a JavaDocs? If it is a comment, it should be 
```
// comment 
or
/* 
text
*/
```

LolfoollorS commented on src/main/java/Command/DeadlineCommand.java:
```diff
@@ -0,0 +1,45 @@
+* Command that adds a Deadline to the TaskList when executed.
+*/
+public class DeadlineCommand extends Command {
+    boolean exit;
```
Perhaps you could rename the variable to make it sound like a boolean.

LolfoollorS commented on src/main/java/Command/DeleteCommand.java:
```diff
@@ -0,0 +1,56 @@
+        this.userAction = userAction;
+    }
+
+    public boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
```
Perhaps you may consider to put it in `execute` via `catch (NumberFormatException e)`

LolfoollorS reviewed (commented):
Overall I think you followed the Java Coding Standard closely.

optionalemon commented on src/main/java/Command/MarkCommand.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    @Override
+    public String toString() {
```
Any reason why you choose to have string functions for command classes? They seem not so required in this chatbot

optionalemon commented on src/main/java/Command/MarkCommand.java:
```diff
@@ -0,0 +1,57 @@
+        this.userAction = userAction;
+    }
+
+    public boolean isNumeric(String str) {
```
perhaps you can put isNumeric in one of the classes and use it, so that you do not have to initialise it everytime in a new class :)

optionalemon commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,52 @@
+                Command c = Parser.parse(fullCommand);
+                if (c != null) {
+                    c.execute(tasks, ui, storage);
+                } else {
```
any reason why you choose to use an else here than putting isExit in if statement?

optionalemon commented on src/main/java/Duke/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+public class DukeException extends Exception {
+
+    public DukeException(String errorMessage) {
+        super(" ☹ OOPS!!! " + errorMessage);
```
I like how an emoji has been put here!

optionalemon commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+                try {
+                    String userAction = command[1];
+                    nextCommand = new UnmarkCommand(userAction);
+                } catch (ArrayIndexOutOfBoundsException e) {
```
Maybe you can put this exception as part of your DukeException! It can be a new user-defined exception extending DukeException :)

linuschui committed:
branch-Level-10

linuschui committed:
branch-Level-10

linuschui committed:
Merge remote-tracking branch 'origin/branch-Level-7'

# Conflicts:
#	tasks.txt

linuschui committed:
Merge tag 'branch-Level-10'

linuschui committed:
Merge remote-tracking branch 'origin/branch-Level-10'

linuschui committed:
Fixed issues with DukeException.

linuschui committed:
DukeException: Allow user to know what commands are available

Duke returns "Invalid command" for any user command that
it does not recognise

Returning the format types of the valid commands will help
the user make readable commands

Let's,
* update message sent when DukeException is thrown to show users
all valid commands types

linuschui committed:
A-Assertions

linuschui committed:
Changed Parser class to check if a description is needed for a command using SLAP.

linuschui committed:
Created a new UpdateCommand to allow users to update task description

linuschui committed:
Merge pull request #2 from linuschui/branch-A-Assertions

A-Assertions

linuschui committed:
A-Assertions

linuschui committed:
Merge branch 'master' into branch-A-CodeQuality

linuschui committed:
Merge pull request #3 from linuschui/branch-A-CodeQuality

A-CodeQuality

linuschui committed:
A-CodeQuality

linuschui committed:
Merge branch 'master' into C-Update

linuschui committed:
Merge pull request #4 from linuschui/C-Update

C-Update

linuschui committed:
C-Update

linuschui committed:
Storage : Abstracted adding and saving tasks functions.

Deep nesting in initial load() and save() functions.
Makes code hard to read

Let's,
* update the load and save functions to abstract
out the checking of type of task to add to TaskList
or save to .txt file

linuschui committed:
A-MoreTesting

linuschui committed:
A-Personality

linuschui committed:
A-BetterGui

linuschui committed:
A-UserGuide

linuschui committed:
Added Ui.png

linuschui committed:
A-Checkstyle

linuschui committed:
Edited README

linuschui committed:
A-Release

linuschui committed:
A-Release
